proposals/gc/MVP-JS.md

```diff
@@ -0,0 +1,532 @@
+that convert to and from arbitrary JavaScript values and the pure typed values
+that WebAssembly is defined to interact with.
+
+The JS Typed Object proposal includes a set of [common Value Type objects](https://github.com/tschneidereit/proposal-typed-objects/blob/master/explainer.md#value-types).
+This general set is extended with additional Value Type objects in the
+`WebAssembly` namespace [described below](#other-webassembly-value-types), so
+that all WebAssembly field types have a corresponding Value Type object. The 
+generally-applicable Value Type objects defined by the JS Typed Object proposal
+are broken into two categories:
+
+* *Primitive* Value Types, exposed as singleton objects: `uint8`,
+  `uint16`, `uint32`, `uint64`, `int8`, `int16`, `int32`, `int64`, `float32`,
+  `float64`, `any`, `string`.
+
+* *Compound* Value Types, which are produced via the `ref` property
+   on another kind of object...
+
+
+### Type Definition objects
+
+Corresponding to the [Type Definitions](https://github.com/WebAssembly/gc/blob/master/proposals/gc/MVP.md#type-definitions)
+in the WebAssembly GC proposal, the JS Typed Objects proposal defines constructors
+for [Type Definition objects](https://github.com/tschneidereit/proposal-typed-objects/blob/master/explainer.md#type-definitions).
+Type Definition objects are constructed by the [`StructType`](https://github.com/tschneidereit/proposal-typed-objects/blob/master/explainer.md#struct-type-definitions)
+and [`ArrayType`](TODO) constructors which are passed Value Type objects for
+each field/element.
+
+Type Definition objects serve three purposes:
+
+* When a type definition is [exported](https://github.com/WebAssembly/gc/blob/master/proposals/gc/MVP.md#exports)
+  by a wasm module, a Type Definition object is produced in the [exportsObject](https://webassembly.github.io/spec/js-api/index.html#instantiate-a-webassembly-module).
+
+* When a type definition is [imported](https://github.com/WebAssembly/gc/blob/master/proposals/gc/MVP.md#imports)
+  by a wasm module, a Type Definition object is expected in the [importsObject](https://webassembly.github.io/spec/js-api/index.html#instantiate-a-webassembly-module).
+
+* A Type Definition object is actually a constructor function which can be
+  called from JS to construct another kind of object...
+
+
+### Typed objects
+
+Corresponding to the [structs](https://github.com/WebAssembly/gc/blob/master/proposals/gc/MVP.md#structures)
+and [arrays](https://github.com/WebAssembly/gc/blob/master/proposals/gc/MVP.md#arrays)
+of the WebAssembly GC proposal, the JS Typed Object proposal defines a new kind of Exotic Object
+called a [typed object](https://github.com/tschneidereit/proposal-typed-objects/blob/master/explainer.md#instantiation).
+Typed objects can be both created from JS by calling a Type Definition object
+as a constructor and created wasm from wasm by executing [`struct.new`](https://github.com/WebAssembly/gc/blob/master/proposals/gc/MVP.md#structures)
```

Comment:
"created wasm from wasm" should be "created from wasm"?

Reply:
Thanks! Fixed.